flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/StaticIcebergEnumerator.java

```diff
@@ -56,30 +46,11 @@ public StaticIcebergEnumerator(
     this.assigner = assigner;
     this.table = table;
     this.scanContext = scanContext;
-    // split enumeration is not needed during restore scenario
-    this.shouldEnumerate = enumState == null;
   }
 
   @Override
   public void start() {
     super.start();
-    if (shouldEnumerate) {
```

Comment:
split discovery for static/batch enumerator is performed in the `IcebergSource` line 183 now. this is to consolidate the batch planning into a single class.